rotkehlchen/db/upgrades/v40_v41.py

```diff
@@ -123,16 +125,61 @@ def _remove_covalent_api_key(write_cursor: 'DBCursor') -> None:
     log.debug('Exit _remove_covalent_api_key')
 
 
+def _move_labels_to_addressbook(write_cursor: 'DBCursor') -> None:
+    """Move all the `label` column values from `blockchain_accounts` table to the `name` column
+    of the 'address_book` table. If a `name` already exists in the `address_book` table, then
+    `address_name_priority` setting is used to determine which one to keep, defaults to
+    `DEFAULT_ADDRESS_NAME_PRIORITY`."""
+    log.debug('Enter _move_labels_to_addressbook')
+    address_name_priority = write_cursor.execute(  # get priority settings
+        'SELECT value FROM settings WHERE name = "address_name_priority"',
+    ).fetchone()
+    if address_name_priority is not None:
+        try:
+            address_name_priority = json.loads(address_name_priority[0])
+        except json.decoder.JSONDecodeError:
+            log.error(
+                'During v40->v41 DB upgrade a non-json address_name_priority setting was found: '
+                f'{address_name_priority[0]}. Reverting to default.',
+            )
+            address_name_priority = DEFAULT_ADDRESS_NAME_PRIORITY
+    else:
+        address_name_priority = DEFAULT_ADDRESS_NAME_PRIORITY
+    address_book_priority = address_name_priority.index('private_addressbook')
+    blockchain_account_priority = address_name_priority.index('blockchain_account')
+    labels_to_move = {  # get all the labels from `blockchain_accounts` table
+        (blockchain, account): label for blockchain, account, label in write_cursor.execute(
+            'SELECT blockchain, account, label FROM blockchain_accounts WHERE label IS NOT NULL',
+        )
+    }
+    if address_book_priority < blockchain_account_priority:
+        for blockchain, address in labels_to_move:
+            already_saved_name = write_cursor.execute(
+                'SELECT name FROM address_book WHERE blockchain=? AND address=?',
+                (blockchain, address),
+            ).fetchone()
+            if already_saved_name is not None:  # replace with `address_book` names
+                labels_to_move[(blockchain, address)] = already_saved_name[0]
```

Comment:
You are checking for existence only right? Use a count(*) instead since is easier to follow. Otherwise it seems that you are missing something